### PR TITLE
Add ExitPortalCreateEvent

### PR DESCRIPTION
**Pull request by ravkr**
This event is called just after calculating position of exit portal in The End, so you can modify its position, set/check if it will be filled with portal blocks etc.

Can be used to place portal higher than default on empty worlds (origin: y=-1; blocks from y=-2 to y=2, so portal was cut in the middle).

### CODE REVIEW

File: Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch
Comment (Machine-Maker) [inline Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: This line, and the one above it can be combined into a single line. `if (!exitPortalEvent.callEvent()) { return; }

File: Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch
Comment (Machine-Maker) [inline Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]:
```suggestion
+        Location location = MCUtil.toLocation(world, exitPortalLocation);
```

File: Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch
Comment (Machine-Maker) [inline Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]:
```suggestion
+        this.exitPortalLocation = MCUtil.toBlockPosition(exitPortalEvent.getPortalLocation());
```

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
just minor things, no functionality changes

File: Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch
Comment (kashike) [inline Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: Imports are not necessary.

**Review by kashike — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch
Comment (ravkr) [reply Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: are you sure? without them it won't compile. (unless I put whole package name before class names) 

edit: other patches use imports too:
https://github.com/PaperMC/Paper/blob/8b47131da76e9a0e4da00d1888950a6e4fa3d07b/Spigot-Server-Patches/0687-added-PlayerNameEntityEvent.patch#L23
https://github.com/PaperMC/Paper/blob/8b47131da76e9a0e4da00d1888950a6e4fa3d07b/Spigot-Server-Patches/0686-Add-worldborder-events.patch#L17

Comment (ravkr) [reply Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: @kashike ?

Comment (aurorasmiles) [reply Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: You can use the full name of functions instead of imports, for example instead of `new ExitPortalCreateEvent(location, flag)` you should be able to use `new io.papermc.paper.event.world.ExitPortalCreateEvent(location, flag)`

Comment (Trigary) [reply Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: Or maybe kashike meant that it's not necessary to add //Paper comments for imports, as maintainers just let their IDEs sort out imports. But who knows for certain what he meant...

Comment (ravkr) [reply Spigot-Server-Patches/0698-Add-ExitPortalCreateEvent.patch]: that's why I'm waiting for his response... almost one month

**Pull request closed without merging**
